HopsanGUI/Dialogs/OptimizationDialog.cpp

```diff
@@ -61,1742 +62,237 @@
-        mpC1LineEdit->setVisible(true);
-        mpC2Label->setVisible(true);
-        mpC2LineEdit->setVisible(true);
-        mpVmaxLabel->setVisible(true);
-        mpVmaxLineEdit->setVisible(true);
-        break;
-    case Ops::DifferentialEvolution:
-        mpFLabel->setVisible(true);
-        mpFLineEdit->setVisible(true);
-        mpCRLabel->setVisible(true);
-        mpCRLineEdit->setVisible(true);
-        break;
-    case Ops::Genetic:
-        mpCPLabel->setVisible(true);
-        mpCPLineEdit->setVisible(true);
-        mpMPLabel->setVisible(true);
-        mpMPLineEdit->setVisible(true);
-    case Ops::ParameterSweep:
-        break;
-    default:
-        break;
-    }
-}
-
-
-//! @brief Adds a new parameter to the list of selected parameter, and displays it in dialog
-//! @param item Tree widget item which represents parameter
-void OptimizationDialog::updateChosenParameters(QTreeWidgetItem* item, int /*i*/)
+//! @brief Updates output boxes displaying the parameters
+void OptimizationDialog::updateParameterOutputs(const std::vector<double> &objectives, const std::vector<std::vector<double> > &values, const int bestId, const int worstId)
 {
-    if(item->checkState(0) == Qt::Checked)
+    if(mOutputDisabled || !this->isVisible()) return;
+
+    if(mParametersOutputLineEditPtrs.size() != (int)objectives.size())
     {
-        mSelectedComponents.append(item->parent()->text(0));
-        mSelectedParameters.append(item->text(0));
-        QString currentValue;
-        if(item->parent()->text(0) == "_System Parameters")
-        {
-            currentValue = mpSystem->getParameterValue(item->text(0));
-        }
-        else
+        recreateParameterOutputLineEdits();
+    }
+
+    mParameterOutputIndexes.clear();
+    mParameterOutputIndexes.append(bestId);
+    mParameterOutputIndexes.append(worstId);
+    if(bestId == worstId)
+    {
+        mParameterOutputIndexes.remove(0);
+    }
+    for(size_t i=0; i<values.size(); ++i)
+    {
+        if(!mParameterOutputIndexes.contains(i))
         {
-            currentValue = mpSystem->getModelObject(item->parent()->text(0))->getParameterValue(item->text(0));
+            mParameterOutputIndexes.append(i);
         }
+    }
 
-        QLabel *pLabel = new QLabel(trUtf8(" <  ") + item->parent()->text(0) + ", " + item->text(0) + " (" + currentValue + trUtf8(")  < "));
-        pLabel->setAlignment(Qt::AlignCenter);
+    for(int x=0; x<mParameterOutputIndexes.size(); ++x)
+    {
+        int i = mParameterOutputIndexes[x];
+        if(i >= (int)objectives.size()) continue;
 
-        OptimizationSettings optSettings;
-        mpSystem->getOptimizationSettings(optSettings);
-        QString min, max;
-        for(int i=0; i<optSettings.mParamters.size(); ++i)
+        QString output = "obj: ";
+        QString objStr;
+        //! @todo This is an error in the code, this must be solved, previous code assumed that i is in range of objects -> ASSERT failed in QVector
```

Comment:
The check is actually redundant already, since it both has the error check and this line above:
if(i >= (int)objectives.size()) continue;
This error will only occur if someone calls the function with objectives and values vectors of different size, which is of course not allowed. And it will not cause a crash anyway, it would just ignore the excessive objectives.